image_captioning/models/encoder/nets/InceptionV3Encoder.py

```diff
@@ -0,0 +1,37 @@
+"""Encoder using the Inception V3 backbone
+"""
+from typing import Iterable, Union
+
+import tensorflow as tf
+
+
+class InceptionV3Encoder(tf.keras.Model):
+    """Encoder using the Inception V3 backbone
+    """
+    def __init__(self, embedding_dim: int):
+        super().__init__()
+        self.embedding_dim = embedding_dim
+
+        # Using Inception V3 backbone
+        self.backbone = tf.keras.applications.InceptionV3(include_top=False,
+                                                          weights='imagenet')
+        new_input = self.backbone.input
+        hidden_layer = self.backbone.layers[-1].output
+        self.backbone = tf.keras.Model(new_input, hidden_layer)
+
+        # FC layer to output into embedding dims
+        self.fc = tf.keras.layers.Dense(embedding_dim)
```

Comment:
It's a personal preference, but have you considered using `tf.keras.Sequential`? Its counterpart of `torch.nn.Sequential` in PyTorch, and you can get the output of all NN modules in a single pass with this

Reply:
No need to use `Sequencial` because I'm using the  in-built Inception network as is.